addons/sale/__init__.py

```diff
@@ -7,9 +7,36 @@
 from . import wizard
 from . import populate
 
+def _post_init_hook(env):
+    _synchronize_cron(env)
+    _setup_property_downpayment_account(env)
+
 
 def _synchronize_cron(env):
     send_invoice_cron = env.ref('sale.send_invoice_cron', raise_if_not_found=False)
     if send_invoice_cron:
         config = env['ir.config_parameter'].get_param('sale.automatic_invoice', False)
         send_invoice_cron.active = bool(config)
+
+
+def _setup_property_downpayment_account(env):
+    property_downpayment_field = env['ir.model.fields']._get("product.category", "property_account_downpayment_categ_id")
+    # Get companies that already have the property set
+    ir_property_companies = env['ir.property'].search([
+        ('name', '=', 'property_account_downpayment_categ_id'),
+    ]).mapped('company_id.id')
+
+    # Create property for companies without it
+    for company in env.companies:
+        if not company.chart_template and company not in ir_property_companies:
+            continue
+
+        template_data = env['account.chart.template']._get_chart_template_data(company.chart_template).get('template_data')
+        if template_data and template_data.get('property_account_downpayment_categ_id'):
+            env['ir.property'].create({
+                'company_id': company.id,
+                'name': 'property_account_downpayment_categ_id',
+                'value_reference': 'account.account,%s' % env.ref(f'account.{company.id}_{template_data["property_account_downpayment_categ_id"]}').id,
```

Comment:
not ultra sexy but not sure there is a better way for that :thinking:

Reply:
can't you factorize that code https://github.com/odoo/odoo/blob/0b8481203caff4dcd7d1ae445007f973ae4126a8/addons/account/models/chart_template.py#L350 in a function and call it here? It seems it is doing what you aim to do, no?